src/query/storages/index/src/range_filter.rs

```diff
@@ -628,6 +628,41 @@ impl<'a> VerifiableExprBuilder<'a> {
                     "Cannot build atom expression by the operator: not like",
                 ))
             }
+            "in" => {
```

Comment:
sort the `in values` may decrease the filter result?   
suppose a filter like `a in [ 0, 10000]`  and a column with min-value 100, max value 1000, this column should be filtered out. but now seems not.

Reply:
I did not get it.

Reply:
@zhyass are we using the “simpler but less tighter falsifiable expression” strategy here?

x IN(𝑐1,...,𝑐𝑁) ~> min(x) > max(𝑐1, ..., 𝑐𝑁 ) OR max(x) < min(𝑐1, ..., 𝑐𝑁 )

if N is not that large, shall we consider the 

x IN(𝑐1,...,𝑐𝑁) ~> (min(x) > 𝑐1 OR max(x) < 𝑐1 ) AND ... AND (min(x)>𝑐 OR max(x)<𝑐 )

Reply:
I probably understand what you mean. 

Now the approach is similar to treating `expr in list` as `expr >= min_list and expr <= max_list`. In fact initially we had considered converting `in` to `eq or` to do it. But this would introduce some other problems.

I'll use an example to illustrate. 
```
a in (1,2,3,4)
```
If we follow the `eq or` approach, the resulting verifiable_expr would be 
```
(min_a <= 1 and max_a >= 1) or (min_a <= 2 and max_a >= 2) or (min_a <= 3 and max_a >= 3) or (min_a <= 4 and max_a >= 4)
``` 
And that's just for the case where the list has 4 values. This will have a bigger impact on performance when doing range prunning. 

So we have to consider some compromise, and there will inevitably be read amplification as you mentioned.

By the way, if the number of inlist is less than 4, the optimizer will convert the `in` to `eq or` before range_filter.

Reply:
> @zhyass are we using the “simpler but less tighter falsifiable expression” strategy here?
> 
> x IN(𝑐1,...,𝑐𝑁) ~> min(x) > max(𝑐1, ..., 𝑐𝑁 ) OR max(x) < min(𝑐1, ..., 𝑐𝑁 )
> 
> if N is not that large, shall we consider the
> 
> x IN(𝑐1,...,𝑐𝑁) ~> (min(x) > 𝑐1 OR max(x) < 𝑐1 ) AND ... AND (min(x)>𝑐 OR max(x)<𝑐 )

@dantengsky This strategy was discussed with @sundy-li before doing it. We need consider the performance impact. On the other hand, if the number of inlist is less than 4, the optimizer will convert the `in` to `eq or` before range_filter. So it may not be necessary to do so.

Reply:
> On the other hand, if the number of inlist is less than 4, the optimizer will convert the in to eq or before range_filter. So it may not be necessary to do so.

yeah, sounds reasonable to me

Reply:
> sort the `in values` may decrease the filter result?
> suppose a filter like `a in [ 0, 10000]` and a column with min-value 100, max value 1000, this column should be filtered out. but now seems not.

@sandflee For this case(number of items lesser than 4),  the tighter form (`eq or`) will be utilized, thus I mark this conversation as resolved. pls feel free to re-open it if any others concerns

Reply:
ok, let's check the actual improvment on the online sql.